src/core/scope.ml

```diff
@@ -603,6 +608,157 @@ let scope_rule : sig_state -> p_rule -> pre_rule loc = fun ss r ->
   in
   Pos.make r.pos pr
 
+(** [scope_hint ss h] transforms a parser-level hint [h] into a rewriting rule
+    defined on {!val:Sign.hint_unif}. A unification hint can be seen as a
+    rewriting rule that rewrite a unification problem into one or more
+    sub-unification problems. *)
+let scope_hint : sig_state -> p_hint -> rule loc = fun ss h ->
+  let (p_l, p_rs) = h.elt in
+  (* NOTE: in the following comments, we consider a unification hint of the
+     form P ≡ Q → x1 ≡ H1, ..., xn ≡ Hn. *)
+  (* Compute the pattern variables. *)
+  let data =
+    (* Get pattern variables of [l] and [r] and verify that they are
+       linear. *)
```

Comment:
Why enforcing linearity?

Reply:
In the following, why duplicating the code of scope_rule and not trying to reuse it instead? Couldn't you see l=r --> t1=u1,..,tn=un as just syntactic sugar for the p_rule (\#unif l r) --> \#list (\#unif t1 u1) .. (\#unif tn un) and modify find_sym function to return the symbols mapped to \#unif and \#list?

Reply:
I think linearity is required to have desirable properties. I should check the paper and proofs, though.

Reply:
Perhaps. But I prefer let this to the user for the moment.

Reply:
And we have some other checks to perform. I think reusability of the code is a real question here: on one hand, duplicating involves more code, but it can be better to duplicate code and specialise it rather than generalise and twist the already existing code for unification hints.

Reply:
Right but I see no reason here. This feature is very experimental. I propose you to do no check at all on unification rules for the moment. Later, if needed, we could add a function for checking some conditions on unification rules, like we do with Sr.check_rule for rewriting rules.

Reply:
No need to modify find_sym after all. And it's better not to do so anyway.

Reply:
OK, in the end we'll have to make a `Sr` module counter part for unification hints.